src/blocks/TextBlock/nodes/ItemLink.astro

```diff
@@ -10,13 +17,23 @@ type Node = {
 };
 type Props = {
   node: Node;
-  link: InternalLinkFragment;
+  link: AnyPage;
 };
 
-const { node, link } = Astro.props;
-const openInNewTab = Boolean(node.meta?.find(entry => entry.id === 'target' && entry.value === '_blank'));
+const { link, node, ...props } = Astro.props;
+const locale = getLocale() as SiteLocale;
+const href = getHref({ locale, record: link });
```

Comment:
Is this a logical place for the `href`, `html` and `link.title` code or should we make a `RecordLink` component somewhere which is then also used by `InlineItem.astro`?

Reply:
@jurgenbelien what do you think about this concept:
- We move this logic to a reusable component in `components/` as it's not a modular block, and the related fragments are in `lib/routing/`.
- We start with a generic `LinkToRecord` component, which we can later add more specific components next to like `LinkToFile`. By the `LinkTo` prefix they automatically stay next to each other: `components/LinkToRecord/LinkToRecord.astro` and `components/LinkToFile/LinkToFile.astro`.
- Should we bundle them in a shared component directory like `components/LinkTo/LinkToRecord.astro` and `LinkToFile.astro` or does this become confusing? Especially since a component like `LinkToRecord/` has its own `README.md`.
- I imagine their APIs becoming `<LinkToRecord record={ ... } ... >` and `<LinkToFile file={ ... } ... >`. Does that make sense to you?

Reply:
Right, yeah that makes sense

Reply:
Regarding the nesting, I'd prefer to keep it as flat as possible. We have around 16 components in the `@components/` folder. I feel that is still a manageable number of items.

Reply:
I've kept the ItemLink component for now. I feel that having the folder `@blocks/TextBlock/nodes` map directly to the nodes in the `Text.astro` components declaration adds clarity.